### DISCUSSION TITLE
DVR session and unreliable internet network

### DISCUSSION THREAD

**Discussion by nums** [OvenMediaLabs/OvenMediaEngine#1454] [Help] [answered]
Some of my clients have lives that last several hours (10 hours sometimes), and must keep the DVR on the entire live. When the connection is stable this does not pose a problem, however if at some point during the live data the connection is cut or the encoder has crashed, the entire past DVR may be lost.

I tried to research these different approaches:
- Set up an RTMP failback by combining several ffmpeg commands (unstable, introducing delay)
- Introduce server-side re-encoding (obviously, resource intensive)

on OvenMedia side, what could be the solutions?

**Comment by naanlizard** [comment]
For our similar situation what we're doing is just using one one instance and caching things with nginx at the edge rather than having an ovt hop, partly because we ran into the same issue

**Reply by nums** [reply]
Thanks for your help @naanlizard, I had thought about it but I was afraid of having conflict issues between two DVR sessions. I will check if my use case could work like this. 
I also made progress on the fallback stream which seems more stable, I will publish the solution if it's worth it.

**Reply by naanlizard** [reply]
Check the discussions tab, specifically mine about edge throughput. It's buried a little but there is cache discussion that might help

**Comment by getroot** [comment, marked as answer]
The Scheduled Channel I am currently developing is a persistent channel. Currently it only uses file as a source, but will be expanded to allow live channels to be used as a source as well. So using this you will be able to create channels that can only be deleted via the API and automatically switch to a prepared file when live input is lost.

**Reply by nums** [reply]
This is very good news indeed.

I created a first POC which works quite well with liquidsoap (which is quite interesting), and provides a fallback method which works for RTMP and SRT (even with 3 audio tracks), however it adds a lot of delay to my stream and random delay (sometimes 5 seconds, sometimes 10 etc)
There is also the concept of passing mpgts packets to an ffmpeg pipeline, I managed to make it work with a stable delay with RTMP.
If anyone is interested I could share the code of these POCs.

But now I'm going to wait for the "scheduled" feature to be finalized and stop my other tests :)

**Comment by getroot** [comment]
@nums 

https://github.com/AirenSoft/OvenMediaEngine/discussions/1461

Check this out, welcome your feedback.

**Reply by nums** [reply]
Yes, thanks, I will test master branch :)
